### ISSUE TITLE
spectator switching/1st person

### ISSUE THREAD

**Issue by CarbonCookies** [yvt/openspades#552]
_This is more of an enhancement_
1. when spectating, sometimes I don't automatically go into the 1st person view, and I can't see any "1st person" button available.
2. after I start spectating, I often have to keep clicking until I reach the person I want, but sometimes I click too much, and I overclock. Could be fixed with arrow keys (right-forward(user# goes up) and left-back(user#goes down)) to switch spectating between different people

**Comment by feikname** [comment]
Maybe adding an option to spectate someone by clicking on them on the Scoreboard would also be an nice feature.

**Comment by CarbonCookies** [comment]
after pressing "esc", a clickable player list come up? (aswell as the other options
)

**Comment by feikname** [comment]
@CarbonCookies more like after pressing whichever key is assigned to the Scoreboard, display it in a clickable manner with additional options. 

Similiar to *Team Fortress 2* with mouse input enabled scoreboard, gut without the nasty bugs that Valve refuses to fix in a long time.

**Comment by CarbonCookies** [comment]
But you need to keep in mind that while in the game, the mouse is always in the middle, where the crosshairs are, so you wouldn't be able to click on much. Ofcourse this could be fixed by locking the crosshairs and allow free movement of the mouse, until either escape, "close", or a players name is clicked.

**Comment by CarbonCookies** [comment]
My idea is something like this: 
![image](https://cloud.githubusercontent.com/assets/24988211/21967517/9c580062-db80-11e6-8274-b4d2a2b05abe.png)
But more professional, using the same fonts as normal, etc. clicking on a link will say "Copied to Clipboard!" , like when it says "Deuce34 has captured the intel". Clicking on an username will close the menu, and start 1st person spectating. Clicking once will go into the free fly mode, clicking again (or using arrows) will cycle through 3rd person perspectives on other people.

However, I'm not sure how hard this would be.

**Comment by feikname** [comment]
I don't approve the GUI the concept you made in the photo above.

Having an separate GUI for the Scoreboard, the Chat Log and the Menu makes them only display the relevant information each one has to offer. 

Trying to display the three of them at the same time, in my opinion, **clutters the menu**. Which also doesn't help much by being in the center of the screen, leaving only the borders for information to be drawn.

There is a reason GitHub and Google have a lot of whitespace on the sides, and it's to make easier for us to focus on the what really matter, which in this case is the *in-game menu*.

With OpenSpades miminum supported resolution, *800x600*, the available horizontal and vertical space is way too small to display an reasonable amount of useful information. And I personally like to play with this resolution in windowed mode, so it's a [*no-no*](http://www.thefreedictionary.com/no-no) to me. 

Displaying the Chat alongside the menu makes no sense to me, its already to possible to see it while *in-game*, and if you have to search for an really older message, just use the Chat Log GUI. 

Implementing [my suggestion of an keyboard shortcut for opening it](https://github.com/yvt/openspades/issues/544#issuecomment-272709088) and 100best's one to [copy text on it and make URLs clickable](https://github.com/yvt/openspades/issues/417) would make it quite easy to access it with in a very practical way.

It'd use the same number of keystrokes required to access the menu, and with way more releavnt information related to what the user wants: the chat log. Same thing for the Scoreboard, with the exception that you need to hold the TAB key instead of pressing it once.

Also, someone would have to basically reimplement the scoreboard and chat windows, while making the menu code unnecessarily more complicatad than it needs to be. To put this much work on it, its easier to just free the crosshair and make the Scoreboard clickable on specator mode by default.

Sorry for the lengthy ranting :/

**TL;DR:** Holding TAB and clicking on a player which will probably be visible from the start sounds better than pressing ESC and scrolling until you find it. Also, re-implementing the Scoreboard and the Chat makes no sense to me.

**Comment by CarbonCookies** [comment]
My bad, I didnt think about that at all. Your idea about TAB opening scoreboard and freeing mouse would be much easier, fully agree on you there.
